src/d3a/models/state.py

```diff
@@ -56,17 +56,33 @@ def get_state(self) -> Dict:
 
     @abstractmethod
     def restore_state(self, state_dict: Dict):
-        pass
+        """Update the state of the device using the provided dictionary."""
 
     @abstractmethod
-    def delete_past_state_values(self, time_slot):
-        pass
+    def delete_past_state_values(self, current_time_slot: DateTime):
+        """Delete the state of the device before the given time slot."""
 
     def __str__(self):
         return self.__class__.__name__
 
 
-class ConsumptionState(StateInterface):
+class ProsumptionInterface(StateInterface, ABC):
+    """Interface with common methods/variables shared by consumption and production devices."""
+
+    def __init__(self):
+        # Actual energy consumed/produced by the device at specific market slots
+        self._energy_measurement_kWh: Dict[DateTime, float] = {}
+
+    def set_energy_measurement_kWh(self, energy_kWh: float, time_slot: DateTime) -> None:
+        """Set the actual energy consumed/produced by the device in the given market slot."""
+        self._energy_measurement_kWh[time_slot] = energy_kWh
+
+    def get_energy_measurement_kWh(self, time_slot: DateTime) -> float:
+        """Get the actual energy consumed/produced by the device in the given market slot."""
+        return self._energy_measurement_kWh.get(time_slot, 0.0)
```

Comment:
This PR is just for simulations without external connection, but I see your point.
However, what should happen when a device doesn't send its measured energy? Don't we apply any penalty? Also, can this scenario happen?

Reply:
>  what should happen when a device doesn't send its measured energy

Thats a matter of regulatory requirement that could be extremely strict or have some relaxation to simply ignore if the device have missed some beats due to network connectivity.

> Don't we apply any penalty?

If later is the case as described in a possible scenario above then the device penalty is simply ignored for some time duration threshold

> Also, can this scenario happen?

I see it quite like to happen quite often at least in the early implementation phase of such systems.

Reply:
It looks like this aspect has not been discussed in the requirements. I'll add a comment on the ticket.

Reply:
From the ticket conversation:
> A None value should be fine, or however a data point is represented if there’s no forecast in our current implementation.
> We can use that for penalties or whatever we decide later as the mechanism.

I decided to use `None` for the moment. Keep in mind that this method is currently not called anywhere: I implemented it because it will be needed later on for the settlement market implementation. In that moment, we need to pay attention to this `None` value (to avoid `TypeError` exceptions).